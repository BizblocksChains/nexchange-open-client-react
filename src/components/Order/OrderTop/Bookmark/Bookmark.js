import React, { Component } from 'react';
import { Modal } from 'react-bootstrap';
import { I18n } from 'react-i18next';
import platform from 'platform';
import CopyToClipboard from 'react-copy-to-clipboard';
import styles from './Bookmark.scss';

class Bookmark extends Component {
  constructor(props) {
    super();

    this.state = {
      show: false,
    };
  }

  triggerCopyTooltip() {
    $('#bookmark .fa-clipboard').tooltip({
      trigger: 'click',
      placement: 'top',
    });

    $('#bookmark .fa-clipboard')
      .tooltip('hide')
      .attr('data-original-title', 'Copied!')
      .tooltip('show');

    setTimeout(() => {
      $('#bookmark .fa-clipboard').tooltip('destroy');
    }, 1000);
  }

  componentDidUpdate() {
    if (this.state.show !== this.props.show) {
      this.setState({
        show: this.props.show,
      });
    }
  }

  render() {
    return (
      <I18n ns="translations">
        {t => (
          <Modal show={this.state.show} onHide={this.props.onClose}>
            <div id="bookmark" className="modal-content">
              <div className="modal-header">
                <button type="button" className="close" data-dismiss="modal" aria-hidden="true" onClick={this.props.onClose}>
                  <i className="material-icons">clear</i>
                </button>
                <h4 className="modal-title" data-toggle="tooltip" data-placement="top" data-original-title="Copy">
                  {t('bookmark.0')}
                </h4>
              </div>

              <div className="modal-body">
                <p className={styles.text}>
                  {t('bookmark.1')} <b>{platform.os.family === 'OS X' ? 'Command (âŒ˜)' : 'Ctrl'} + D</b> {t('bookmark.2')}
                </p>

                <div className="input-group">
                  <div className="form-group is-empty">
                    <input type="text" className="form-control" placeholder="With Material Icons" value={window.location.href} disabled />
                    <span className="material-input" />
                  </div>

                  <CopyToClipboard text={window.location.href} onCopy={() => this.triggerCopyTooltip()}>
                    <span className={`${styles['input-group-addon']} input-group-addon`}>
                      <i className="fas fa-clipboard" aria-hidden="true" />
                    </span>
                  </CopyToClipboard>
                </div>
              </div>
            </div>
          </Modal>
        )}
      </I18n>
    );
  }
}

export default Bookmark;
