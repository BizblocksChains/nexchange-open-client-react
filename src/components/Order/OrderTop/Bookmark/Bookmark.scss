.bookmark {
  font-size: 19px;
}

.input-group-addon {
  cursor: pointer;
  padding-top: 0;

  i {
    font-size: 18px;
    color: $blue;
  }
}
