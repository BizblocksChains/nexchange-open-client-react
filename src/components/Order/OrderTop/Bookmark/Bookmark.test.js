import React from 'react';
import { shallow } from 'enzyme';
import Bookmark from './Bookmark';

describe('Bookmark', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(<Bookmark />).dive();
    expect(wrapShallow).toMatchSnapshot();
  });
});
