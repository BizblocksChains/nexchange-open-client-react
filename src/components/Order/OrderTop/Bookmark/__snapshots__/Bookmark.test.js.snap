// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animation": true,
      "autoFocus": true,
      "backdrop": true,
      "bsClass": "modal",
      "children": <div
        className="modal-content"
        id="bookmark"
>
        <div
                className="modal-header"
        >
                <button
                        aria-hidden="true"
                        className="close"
                        data-dismiss="modal"
                        onClick={undefined}
                        type="button"
                >
                        <i
                                className="material-icons"
                        >
                                clear
                        </i>
                </button>
                <h4
                        className="modal-title"
                        data-original-title="Copy"
                        data-placement="top"
                        data-toggle="tooltip"
                >
                        bookmark.0
                </h4>
        </div>
        <div
                className="modal-body"
        >
                <p
                        className="text"
                >
                        bookmark.1
                         
                        <b>
                                Ctrl
                                 + D
                        </b>
                         
                        bookmark.2
                </p>
                <div
                        className="input-group"
                >
                        <div
                                className="form-group is-empty"
                        >
                                <input
                                        className="form-control"
                                        disabled={true}
                                        placeholder="With Material Icons"
                                        type="text"
                                        value="http://localhost/"
                                />
                                <span
                                        className="material-input"
                                />
                        </div>
                        <CopyToClipboard
                                onCopy={[Function]}
                                options={undefined}
                                text="http://localhost/"
                        >
                                <span
                                        className="input-group-addon input-group-addon"
                                >
                                        <i
                                                aria-hidden="true"
                                                className="fas fa-clipboard"
                                        />
                                </span>
                        </CopyToClipboard>
                </div>
        </div>
</div>,
      "dialogComponentClass": [Function],
      "enforceFocus": true,
      "keyboard": true,
      "manager": ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      },
      "onHide": [Function],
      "renderBackdrop": [Function],
      "restoreFocus": true,
      "show": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modal-header"
>
            <button
                        aria-hidden="true"
                        className="close"
                        data-dismiss="modal"
                        onClick={undefined}
                        type="button"
            >
                        <i
                                    className="material-icons"
                        >
                                    clear
                        </i>
            </button>
            <h4
                        className="modal-title"
                        data-original-title="Copy"
                        data-placement="top"
                        data-toggle="tooltip"
            >
                        bookmark.0
            </h4>
</div>,
          <div
            className="modal-body"
>
            <p
                        className="text"
            >
                        bookmark.1
                         
                        <b>
                                    Ctrl
                                     + D
                        </b>
                         
                        bookmark.2
            </p>
            <div
                        className="input-group"
            >
                        <div
                                    className="form-group is-empty"
                        >
                                    <input
                                                className="form-control"
                                                disabled={true}
                                                placeholder="With Material Icons"
                                                type="text"
                                                value="http://localhost/"
                                    />
                                    <span
                                                className="material-input"
                                    />
                        </div>
                        <CopyToClipboard
                                    onCopy={[Function]}
                                    options={undefined}
                                    text="http://localhost/"
                        >
                                    <span
                                                className="input-group-addon input-group-addon"
                                    >
                                                <i
                                                            aria-hidden="true"
                                                            className="fas fa-clipboard"
                                                />
                                    </span>
                        </CopyToClipboard>
            </div>
</div>,
        ],
        "className": "modal-content",
        "id": "bookmark",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                aria-hidden="true"
                className="close"
                data-dismiss="modal"
                onClick={undefined}
                type="button"
>
                <i
                                className="material-icons"
                >
                                clear
                </i>
</button>,
              <h4
                className="modal-title"
                data-original-title="Copy"
                data-placement="top"
                data-toggle="tooltip"
>
                bookmark.0
</h4>,
            ],
            "className": "modal-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "children": <i
                  className="material-icons"
>
                  clear
</i>,
                "className": "close",
                "data-dismiss": "modal",
                "onClick": undefined,
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "clear",
                  "className": "material-icons",
                },
                "ref": null,
                "rendered": "clear",
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "bookmark.0",
                "className": "modal-title",
                "data-original-title": "Copy",
                "data-placement": "top",
                "data-toggle": "tooltip",
              },
              "ref": null,
              "rendered": "bookmark.0",
              "type": "h4",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="text"
>
                bookmark.1
                 
                <b>
                                Ctrl
                                 + D
                </b>
                 
                bookmark.2
</p>,
              <div
                className="input-group"
>
                <div
                                className="form-group is-empty"
                >
                                <input
                                                className="form-control"
                                                disabled={true}
                                                placeholder="With Material Icons"
                                                type="text"
                                                value="http://localhost/"
                                />
                                <span
                                                className="material-input"
                                />
                </div>
                <CopyToClipboard
                                onCopy={[Function]}
                                options={undefined}
                                text="http://localhost/"
                >
                                <span
                                                className="input-group-addon input-group-addon"
                                >
                                                <i
                                                                aria-hidden="true"
                                                                className="fas fa-clipboard"
                                                />
                                </span>
                </CopyToClipboard>
</div>,
            ],
            "className": "modal-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "bookmark.1",
                  " ",
                  <b>
                    Ctrl
                     + D
</b>,
                  " ",
                  "bookmark.2",
                ],
                "className": "text",
              },
              "ref": null,
              "rendered": Array [
                "bookmark.1",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Ctrl",
                      " + D",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Ctrl",
                    " + D",
                  ],
                  "type": "b",
                },
                " ",
                "bookmark.2",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="form-group is-empty"
>
                    <input
                                        className="form-control"
                                        disabled={true}
                                        placeholder="With Material Icons"
                                        type="text"
                                        value="http://localhost/"
                    />
                    <span
                                        className="material-input"
                    />
</div>,
                  <CopyToClipboard
                    onCopy={[Function]}
                    options={undefined}
                    text="http://localhost/"
>
                    <span
                                        className="input-group-addon input-group-addon"
                    >
                                        <i
                                                            aria-hidden="true"
                                                            className="fas fa-clipboard"
                                        />
                    </span>
</CopyToClipboard>,
                ],
                "className": "input-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="form-control"
                        disabled={true}
                        placeholder="With Material Icons"
                        type="text"
                        value="http://localhost/"
/>,
                      <span
                        className="material-input"
/>,
                    ],
                    "className": "form-group is-empty",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-control",
                        "disabled": true,
                        "placeholder": "With Material Icons",
                        "type": "text",
                        "value": "http://localhost/",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "material-input",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <span
                      className="input-group-addon input-group-addon"
>
                      <i
                                            aria-hidden="true"
                                            className="fas fa-clipboard"
                      />
</span>,
                    "onCopy": [Function],
                    "options": undefined,
                    "text": "http://localhost/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <i
                        aria-hidden="true"
                        className="fas fa-clipboard"
/>,
                      "className": "input-group-addon input-group-addon",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-hidden": "true",
                        "className": "fas fa-clipboard",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    "type": "span",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": true,
        "autoFocus": true,
        "backdrop": true,
        "bsClass": "modal",
        "children": <div
          className="modal-content"
          id="bookmark"
>
          <div
                    className="modal-header"
          >
                    <button
                              aria-hidden="true"
                              className="close"
                              data-dismiss="modal"
                              onClick={undefined}
                              type="button"
                    >
                              <i
                                        className="material-icons"
                              >
                                        clear
                              </i>
                    </button>
                    <h4
                              className="modal-title"
                              data-original-title="Copy"
                              data-placement="top"
                              data-toggle="tooltip"
                    >
                              bookmark.0
                    </h4>
          </div>
          <div
                    className="modal-body"
          >
                    <p
                              className="text"
                    >
                              bookmark.1
                               
                              <b>
                                        Ctrl
                                         + D
                              </b>
                               
                              bookmark.2
                    </p>
                    <div
                              className="input-group"
                    >
                              <div
                                        className="form-group is-empty"
                              >
                                        <input
                                                  className="form-control"
                                                  disabled={true}
                                                  placeholder="With Material Icons"
                                                  type="text"
                                                  value="http://localhost/"
                                        />
                                        <span
                                                  className="material-input"
                                        />
                              </div>
                              <CopyToClipboard
                                        onCopy={[Function]}
                                        options={undefined}
                                        text="http://localhost/"
                              >
                                        <span
                                                  className="input-group-addon input-group-addon"
                                        >
                                                  <i
                                                            aria-hidden="true"
                                                            className="fas fa-clipboard"
                                                  />
                                        </span>
                              </CopyToClipboard>
                    </div>
          </div>
</div>,
        "dialogComponentClass": [Function],
        "enforceFocus": true,
        "keyboard": true,
        "manager": ModalManager {
          "add": [Function],
          "containers": Array [],
          "data": Array [],
          "handleContainerOverflow": true,
          "hideSiblingNodes": true,
          "isTopModal": [Function],
          "modals": Array [],
          "remove": [Function],
        },
        "onHide": [Function],
        "renderBackdrop": [Function],
        "restoreFocus": true,
        "show": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="modal-header"
>
              <button
                            aria-hidden="true"
                            className="close"
                            data-dismiss="modal"
                            onClick={undefined}
                            type="button"
              >
                            <i
                                          className="material-icons"
                            >
                                          clear
                            </i>
              </button>
              <h4
                            className="modal-title"
                            data-original-title="Copy"
                            data-placement="top"
                            data-toggle="tooltip"
              >
                            bookmark.0
              </h4>
</div>,
            <div
              className="modal-body"
>
              <p
                            className="text"
              >
                            bookmark.1
                             
                            <b>
                                          Ctrl
                                           + D
                            </b>
                             
                            bookmark.2
              </p>
              <div
                            className="input-group"
              >
                            <div
                                          className="form-group is-empty"
                            >
                                          <input
                                                        className="form-control"
                                                        disabled={true}
                                                        placeholder="With Material Icons"
                                                        type="text"
                                                        value="http://localhost/"
                                          />
                                          <span
                                                        className="material-input"
                                          />
                            </div>
                            <CopyToClipboard
                                          onCopy={[Function]}
                                          options={undefined}
                                          text="http://localhost/"
                            >
                                          <span
                                                        className="input-group-addon input-group-addon"
                                          >
                                                        <i
                                                                      aria-hidden="true"
                                                                      className="fas fa-clipboard"
                                                        />
                                          </span>
                            </CopyToClipboard>
              </div>
</div>,
          ],
          "className": "modal-content",
          "id": "bookmark",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  aria-hidden="true"
                  className="close"
                  data-dismiss="modal"
                  onClick={undefined}
                  type="button"
>
                  <i
                                    className="material-icons"
                  >
                                    clear
                  </i>
</button>,
                <h4
                  className="modal-title"
                  data-original-title="Copy"
                  data-placement="top"
                  data-toggle="tooltip"
>
                  bookmark.0
</h4>,
              ],
              "className": "modal-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "children": <i
                    className="material-icons"
>
                    clear
</i>,
                  "className": "close",
                  "data-dismiss": "modal",
                  "onClick": undefined,
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "clear",
                    "className": "material-icons",
                  },
                  "ref": null,
                  "rendered": "clear",
                  "type": "i",
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "bookmark.0",
                  "className": "modal-title",
                  "data-original-title": "Copy",
                  "data-placement": "top",
                  "data-toggle": "tooltip",
                },
                "ref": null,
                "rendered": "bookmark.0",
                "type": "h4",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="text"
>
                  bookmark.1
                   
                  <b>
                                    Ctrl
                                     + D
                  </b>
                   
                  bookmark.2
</p>,
                <div
                  className="input-group"
>
                  <div
                                    className="form-group is-empty"
                  >
                                    <input
                                                      className="form-control"
                                                      disabled={true}
                                                      placeholder="With Material Icons"
                                                      type="text"
                                                      value="http://localhost/"
                                    />
                                    <span
                                                      className="material-input"
                                    />
                  </div>
                  <CopyToClipboard
                                    onCopy={[Function]}
                                    options={undefined}
                                    text="http://localhost/"
                  >
                                    <span
                                                      className="input-group-addon input-group-addon"
                                    >
                                                      <i
                                                                        aria-hidden="true"
                                                                        className="fas fa-clipboard"
                                                      />
                                    </span>
                  </CopyToClipboard>
</div>,
              ],
              "className": "modal-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "bookmark.1",
                    " ",
                    <b>
                      Ctrl
                       + D
</b>,
                    " ",
                    "bookmark.2",
                  ],
                  "className": "text",
                },
                "ref": null,
                "rendered": Array [
                  "bookmark.1",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Ctrl",
                        " + D",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Ctrl",
                      " + D",
                    ],
                    "type": "b",
                  },
                  " ",
                  "bookmark.2",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="form-group is-empty"
>
                      <input
                                            className="form-control"
                                            disabled={true}
                                            placeholder="With Material Icons"
                                            type="text"
                                            value="http://localhost/"
                      />
                      <span
                                            className="material-input"
                      />
</div>,
                    <CopyToClipboard
                      onCopy={[Function]}
                      options={undefined}
                      text="http://localhost/"
>
                      <span
                                            className="input-group-addon input-group-addon"
                      >
                                            <i
                                                                  aria-hidden="true"
                                                                  className="fas fa-clipboard"
                                            />
                      </span>
</CopyToClipboard>,
                  ],
                  "className": "input-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          className="form-control"
                          disabled={true}
                          placeholder="With Material Icons"
                          type="text"
                          value="http://localhost/"
/>,
                        <span
                          className="material-input"
/>,
                      ],
                      "className": "form-group is-empty",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "form-control",
                          "disabled": true,
                          "placeholder": "With Material Icons",
                          "type": "text",
                          "value": "http://localhost/",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "material-input",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "span",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <span
                        className="input-group-addon input-group-addon"
>
                        <i
                                                aria-hidden="true"
                                                className="fas fa-clipboard"
                        />
</span>,
                      "onCopy": [Function],
                      "options": undefined,
                      "text": "http://localhost/",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <i
                          aria-hidden="true"
                          className="fas fa-clipboard"
/>,
                        "className": "input-group-addon input-group-addon",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "aria-hidden": "true",
                          "className": "fas fa-clipboard",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                      "type": "span",
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
